### PR TITLE
Added js code review checklist

### PR DESCRIPTION
**Pull request by orangejenny**
## Technical Summary
This follows up on the knowledge sharing session on front end code review. Content comes from https://docs.google.com/presentation/d/1ecYuckbFdBxcO-1lFeLnXgPEzLF1EzhW2pTFHgTrQYQ/edit#slide=id.g16e9478f934_0_172

Added to the JS guide because this is HQ specific content but our [main code review docs](https://github.com/dimagi/open-source/) are not.

## Safety Assurance

Docs only.

### CODE REVIEW

File: docs/js-guide/code-review.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+----------------
+
+Again, standards in JavaScript are largely the same as in Python. However, there are a few issues that are either
+specific to JavaScript or more frequently arise in it:
+
+- Naming. JavaScript is often messy because it sometimes uses server naming conventions, which are different, for server
+  data. Push the author to leave the code better than they found it. Don't allow the same identifier to be used
+  with different capitalizations.
+- JavaScript should be enclosed in modules and those modules should explicitly declare dependencies, as in the
+  first code block here. Exceptions are app manager, reports, and web apps.
```
Comment (millerdev) [inline docs/js-guide/code-review.rst:69]: What is "the first code block here" referring to?

**Review by millerdev — APPROVED** [review]

File: docs/js-guide/code-review.rst (hunk 1 above)
Comment (orangejenny) [reply docs/js-guide/code-review.rst]: oops, thanks - ca30e32ff3e

File: docs/js-guide/code-review.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+Make any user-facing language as clear as possible.
+
+- Proofread it.
+- Limit jargon and overly technical language (using pre-existing HQ terms is okay)
+- Don't let internal names bleed into user-facing content
+
+   - "Lookup tables" not "fixtures"
+   - "Web apps" not "cloudcare"
+   - "Mobile worker" not "mobile user"
```
Comment (mkangia) [inline docs/js-guide/code-review.rst:21]: nit: not "CommCare User"?

File: docs/js-guide/code-review.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+----------
+
+- All user-facing dates and times should be displayed in a time zone that will make sense for the user. Look at
+  usage of ``UserTime`` and more generally at ``corehq.util.timezones``.
+
+Security
+--------
+
+- Use ``<%- ... %>`` in Underscore templates to HTML escape values.
+- Use ``DomPurify`` to HTML escape user input that will displayed, but not in a template.
```
Comment (mkangia) [inline docs/js-guide/code-review.rst:41]: nit: will :be" displayed.

File: docs/js-guide/code-review.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+Coding Standards
+----------------
+
+Again, standards in JavaScript are largely the same as in Python. However, there are a few issues that are either
+specific to JavaScript or more frequently arise in it:
+
+- Naming. JavaScript is often messy because it sometimes uses server naming conventions, which are different, for server
+  data. Push the author to leave the code better than they found it. Don't allow the same identifier to be used
+  with different capitalizations.
```
Comment (mkangia) [inline docs/js-guide/code-review.rst:67]: > Don't allow the same identifier to be used with different capitalizations

What is the recommendation here? same identifier should be used instead?

File: docs/translations.rst [hunk 5]
```diff
@@ -36,7 +36,7 @@ appropriate.
           "fra": {"hello": "bonjour"},
       }
 
-      def dumb_gettext(str):
+      def naive_gettext(str):
```
Comment (mkangia) [inline docs/translations.rst:39]: :white_heart:

**Comment by orangejenny** [comment]
@mkangia Thank you for the feedback, updated in c75d84df8b5

**Comment by mkangia** [comment]
Thank you @orangejenny

File: docs/js-guide/code-review.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+Coding Standards
+----------------
+
+Again, standards in JavaScript are largely the same as in Python. However, there are a few issues that are either
+specific to JavaScript or more frequently arise in it:
+
+- Naming. JavaScript is often messy because it sometimes uses server naming conventions, which are different, for server
+  data. Push the author to leave the code better than they found it. Don't allow the same identifier to be used
+  with different capitalizations, e.g., ``firstName`` and ``first_name`` in the same file. Find a synonym for one
+  of them.
```
Comment (esoergel) [inline docs/js-guide/code-review.rst:68]: Is this allowed for javascript that wraps or interprets an object sent from the server?  Like:
```javascript
this.firstName = data.first_name
```

Comment (orangejenny) [reply docs/js-guide/code-review.rst:68]: We don't have lint rules for js object properties, so you can do what you want. My impression is that rename code like you wrote exists but that in most places we keep the server style.

**Pull request merged**